### PR TITLE
Feature/opencms data loader

### PR DESCRIPTION
**Pull request by romanoe**
data loader component for opencms integration : 
* with filter for preferred and published RIA: #18 REIA-Web
* using originid parameter : [#15](https://github.com/swiss-seismological-service/REIA-ws/issues/15) REIA-ws

### CODE REVIEW

File: src/js/webcomponents/DataLoader.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
+/* eslint-disable max-len */
```
Comment (schmidni) [inline src/js/webcomponents/DataLoader.js:1]: This is not a solution, we need to agree on formatting :)

File: src/js/utils/api.js [hunk 2]
```diff
@@ -39,10 +39,11 @@ export function getRiskAssessment(oid, apiAddress = null) {
     return getData(`/v1/riskassessment/${oid}`, apiAddress);
 }
 
-export function getAllRiskAssessments(limit = 20, offset = 0, apiAddress = null) {
-    return getData(`/v1/riskassessment?limit=${limit}&offset=${offset}`, apiAddress);
+export function getAllRiskAssessments(limit = 20, offset = 0, originid = null, apiAddress = null) {
+    return getData(`/v1/riskassessment?limit=${limit}&offset=${offset}&originid=${originid}`, apiAddress);
```
[the remark this answers is not in the document]
Comment (romanoe) [reply src/js/utils/api.js]: indeed, I add the originid only if (originid). see new commit

File: src/js/webcomponents/DataLoader.js (hunk 1 above)
Comment (romanoe) [reply src/js/webcomponents/DataLoader.js]: yep sure, vscode settings solved 😄

File: src/js/webcomponents/DataLoader.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+import { getAllRiskAssessments, 
+         getLoss, 
+         getCantonalInjuries, 
+         getCantonalStructuralDamage } from "../utils/api";
+
+class DataLoader extends HTMLElement {
+  
+    static get observedAttributes() {
```
[the remark this answers is not in the document]
Comment (romanoe) [reply src/js/webcomponents/DataLoader.js]: indeed, new commit should make this webcomponent reactive and therefore these attributes are actually observed

File: src/js/webcomponents/DataLoader.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+                    // Loss graph
+                    document.querySelector('loss-graph')
+                            .setData(getCantonalInjuries(lossCalculationOid, baseUrl));
+
+                    // Damage graph
+                    document.querySelector('damage-graph')
+                            .setData(getCantonalStructuralDamage(damageCalculationOid, baseUrl));
+
+                } else if (data.length > 1) {
```
[the remark this answers is not in the document]
Comment (romanoe) [reply src/js/webcomponents/DataLoader.js]: true. this condition has been removed.

File: src/js/webcomponents/DataLoader.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+         getCantonalInjuries, 
+         getCantonalStructuralDamage } from "../utils/api";
+
+class DataLoader extends HTMLElement {
+  
+    static get observedAttributes() {
+        return ['originId', 'baseUrl'];
+    }
+
+    connectedCallback() {
```
[the remark this answers is not in the document]
Comment (romanoe) [reply src/js/webcomponents/DataLoader.js]: new implementation should make the component reactive

File: src/js/webcomponents/DataLoader.js [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+        const originId = this.getAttribute('originId');
+        const baseUrl = this.getAttribute('baseUrl');
+
+        if (!originId || !baseUrl) {
+            console.error('Missing required attributes: originId or baseUrl');
+            return;
+        }
+
+
+        const fetchRiskAssessmentData = async () => {
```
[the remark this answers is not in the document]
Comment (romanoe) [reply src/js/webcomponents/DataLoader.js]: done in the new commit

File: src/js/webcomponents/DataLoader.js [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+class DataLoader extends HTMLElement {
+  
+    static get observedAttributes() {
+        return ['originId', 'baseUrl'];
+    }
+
+    connectedCallback() {
+        const originId = this.getAttribute('originId');
+        const baseUrl = this.getAttribute('baseUrl');
```
[the remark this answers is not in the document]
Comment (romanoe) [reply src/js/webcomponents/DataLoader.js]: done in the new commit

File: src/js/webcomponents/DataLoader.js [hunk 8]
```diff
@@ -0,0 +1,91 @@
+/* eslint-disable max-len */
+import { getAllRiskAssessments, 
+         getLoss, 
+         getCantonalInjuries, 
+         getCantonalStructuralDamage } from "../utils/api";
+
+class DataLoader extends HTMLElement {
+  
+    static get observedAttributes() {
+        return ['originId', 'baseUrl'];
```
[the remark this answers is not in the document]
Comment (romanoe) [reply src/js/webcomponents/DataLoader.js]: indeed. the attributes are now lowercase

File: src/js/webcomponents/DataLoader.js [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+                        );
+                    };
+
+                    // Loss scales
+                    document.querySelectorAll('loss-scale')
+                            .forEach(setLossData);
+
+                    // Loss graph
+                    document.querySelector('loss-graph')
+                            .setData(getCantonalInjuries(lossCalculationOid, baseUrl));
```
[the remark this answers is not in the document]
Comment (romanoe) [reply src/js/webcomponents/DataLoader.js]: I don't know why it worked. In the new changes, I access the first element of the array which is returned from the `fetchRiskAssessmentData()` method

File: src/js/webcomponents/DataLoader.js [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+    updateDamageGraph(damageCalculationOid) {
+        if (this.damageGraph) {
+            this.damageGraph.setData(
+                getCantonalStructuralDamage(damageCalculationOid, this.baseurl)
+            );
+        }
+    }
+
+    setLossData(lossScale, lossScaleCategory, lossCalculationOid) {
```
Comment (schmidni) [inline src/js/webcomponents/DataLoader.js:73]: IMO its not necessary to have this as a separate function in this implementation. This way `updateData` calls `updateLossScales` calls `setLossData` which is a bit confusing if not required...

**Comment by schmidni** [comment]
apart from the little comment (which you can ignore or not) its fine and you can merge

**Pull request merged**